conda/conda-index
# Write sharded repodata

### Description

What would it look like to generate sharded repodata per https://github.com/conda/ceps/pull/75

Interested in seeing whether we can efficiently generate shards; how repodata patching should work; and whether we could generate shards as the primary artifact and then derive repodata.json from shards [at the same time in "processes a sequence of package names" code]

How to test

Check out this repository and https://github.com/dholth/conda-test-data

Decompress `conda_test_data/conda-forge/*/.cache/cache.db.zst`

Download `conda-forge-repodata-patches-<version>.conda`

`python3 -m conda_index --write-shards --no-write-monolithic --upstream-stage=clone --no-update-cache --patch-generator ~/miniconda3/pkgs/conda-forge-repodata-patches-20240401.20.33.07-hd8ed1ab_1.conda  --output /tmp/shards ~/prog/conda-test-data/conda-forge`

Examine output in `/tmp/shards`

I've begun trying to apply the patches to individual shards. This is slow; should compare against applying the many patches against a whole repodata.json.

### Checklist - did you ...



- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
